V3 licenses resources

russbiggs opened the pull request:
This add the `v3/licenses` and `v3/licenses/{licenses_id}` endpoints. Current incomplete it modified how licenses are represented in the location and porviders resource which will require updates to the upstream postgres views that feed the queries

jflasher commented:
Thanks!

russbiggs committed:
added licenses resources

russbiggs committed:
license resources

russbiggs commented:
see https://github.com/openaq/openaq-db/pull/31 for updates to locations_view to reflect these changes.

caparker committed:
Added validation and some tests, removed trailing spaces

caparker commented:
I added some tests to this and did some related work on the db. We just need to add the resources and then this one will be ready right?

russbiggs commented:
Only change I would suggest is moving the test_utils to the unit directory, idea being we keep all unit tests in there and everything else that looks more integration, e2e like goes in the top level `tests`

caparker committed:
Merge branch 'v3-release' into v3-licenses-resources

caparker committed:
Added license router to main app

  - Moved the utils tests to the unit tests
  - remove locations from the url list because its covered in the
    sensors test now

caparker reviewed (changes requested):
I added `/v3/licenses` and '/v3/license/1` paths to our tests and neither of them are passing
```
FAILED tests/test_sensors.py::TestUrls::test_urls[url14] - fastapi.exceptions.ResponseValidationError
FAILED tests/test_sensors.py::TestUrls::test_urls[url15] - AssertionError: assert 500 == 200
```
@russbiggs 
I have not dug in any deeper than that, could you look into it?

russbiggs committed:
fixes missing columns for licenses

russbiggs committed:
license resource updates

caparker committed:
Just some spacing my editor fixed
